core/src/test/scala/org/apache/spark/scheduler/BlacklistTrackerSuite.scala

```diff
@@ -456,4 +461,69 @@ class BlacklistTrackerSuite extends SparkFunSuite with BeforeAndAfterEach with M
       conf.remove(config)
     }
   }
+
+  test("blacklisting kills executors, configured by BLACKLIST_KILL_ENABLED") {
+    val allocationClientMock = mock[ExecutorAllocationClient]
+    when(allocationClientMock.killExecutors(any(), any(), any())).thenReturn(Seq("called"))
+    when(allocationClientMock.killExecutorsOnHost("hostA")).thenAnswer(new Answer[Boolean] {
+      override def answer(invocation: InvocationOnMock): Boolean = {
+        if (blacklist.nodeBlacklist.contains("hostA") == false) {
+          throw new IllegalStateException("hostA should be on the blacklist")
```

Comment:
Worth adding a comment here explaining what the point of this check is

To avoid a race between blacklisting and killing, its important that the nodeBlacklist is updated *before* we ask the executor allocation client to kill executors on the host.